lib/pdf/reader/error.rb

```diff
@@ -53,6 +53,10 @@ def self.assert_equal(lvalue, rvalue)
   class MalformedPDFError < RuntimeError; end
 
   ################################################################################
+  # an exception that is raised when an invalid page number is used
+  class InvalidPageError < RuntimeError; end
```

Comment:
This new error is generally raised when someone requests an invalid page number - are you open to changing this to inherit from `ArgumentError`?

Reply:
Yes, good point